data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/action/OneWayHashAction.java

```diff
@@ -0,0 +1,118 @@
+
+    @DataPrepperPluginConstructor
+    public OneWayHashAction(final OneWayHashActionConfig config) {
+        
+        this.salt = config.getSalt();
+        this.saltKey = config.getSaltKey();
+        try{
+            messageDigest = MessageDigest.getInstance(config.getHashFormat());            
+        } catch (NoSuchAlgorithmException noSuchAlgorithmException){
+            LOG.error("There was an exception while processing Event [{}]", noSuchAlgorithmException);
+        }
+    }
+
+    @Override
+    public String obfuscate(String source, List<Pattern> patterns, Record<Event> record) {
+
+        byte [] saltToApply = this.salt;
+                
+        // Resolve salt to compute based on a path provided in the configuration.  
+        // For records where path was not found, the salt value defined in the pipeline configuration will be used, if salt value was not configured, one will be generated.        
+
+        if(saltKey != null && saltKey.equals("") == false){                
+            final Event recordEvent = record.getData();                        
+            if(recordEvent.containsKey(saltKey)){
+                saltToApply = computeSaltBasedOnKeyValue(recordEvent.get(saltKey, String.class));
+            } else {
+                LOG.info("Unable to find a key '{}' for using as salt, using default salt pipeline configuration for the record instead", saltKey);
+            }         
+        }
+
+        if (patterns == null || patterns.size() == 0) {
+            // no pattern to match, replace the whole string
+            return oneWayHashString(source,saltToApply);
+        }
+        
+        String replacementString = source;
+
+        for (Pattern pattern : patterns) {
+            Matcher matcher = Pattern.compile(pattern.pattern()).matcher(replacementString);
+            StringBuffer stringBuffer = new StringBuffer();
+            while (matcher.find()) {
+
+                String stringToHash = source.substring(matcher.start(),matcher.end());
+
+                matcher.appendReplacement(stringBuffer, oneWayHashString(stringToHash,saltToApply));
+            }
+            matcher.appendTail(stringBuffer);
+            replacementString = stringBuffer.toString();
+        }
+        return replacementString;
+        
+        
+    }
+
+    private String oneWayHashString(String source, byte[] salt){
+        String oneWayHashedSource = "";
+
+        try {
+            MessageDigest messageDigestClone = (MessageDigest) messageDigest.clone();
+
+            messageDigestClone.update(salt);
+            byte[] bytes = messageDigestClone.digest(source.getBytes(StandardCharsets.UTF_8));
+            
+            oneWayHashedSource =  Base64.getEncoder().encodeToString(bytes);
+
+        } catch (CloneNotSupportedException cloneNotSupportedException){
+            LOG.error("There was an exception while processing Event [{}]", cloneNotSupportedException);
```

Comment:
I think you should re-throw this exception. Then we should let the `ObfuscateProcessor` handle exceptions from the actions. This way, it can be done consistently, perhaps with tagging. 

```
throw new RuntimeException(cloneNotSupportedException);
```

What do you think the correct behavior should be when we fail to obfuscate data? Should we drop the message? I wonder if we should support an option for that or let the user tag. Also, you don't have to solve that in this PR, but let's design for this by throwing the exception.

At least for now, re-throw to the processor.

Reply:
Made the change.